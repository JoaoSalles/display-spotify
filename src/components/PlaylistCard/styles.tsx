import styled from "styled-components";
import { BLACK, GRAY, BORDER_WHITE } from "constants/colors";

export const PlaylistCardElement = styled.a`
  &.playlist-card {
    transition: transform 0.2s;
    transition-timing-function: ease-in;
    align-items: center;
    color: ${BLACK};
    display: flex;
    flex: 0 23%;
    flex-direction: column;
    height: 100%;
    text-decoration: none;
    position: relative;
    width: 100%;
    margin: 10px 1%;

    .playlist-card__content {
      border: 1px solid ${BORDER_WHITE};
      border-radius: 5px;
      box-shadow: 0px 1px 4px rgba(0, 0, 0, 0.05);
      text-align: center;

      &__description {
        display: inline-grid;

        &__title {
          font-weight: 600;
        }

        &__tracks {
          color: ${GRAY};
        }
      }
    }

    img {
      border-radius: 5px;
      max-height: 100%;
      max-width: 100%;
    }

    &:hover {
      transform: translateY(-5px);
    }

    @media only screen and (max-width: 1439px) {
      flex: 0 31%;
    }

    @media only screen and (max-width: 768px) {
      flex: 0 47%;
    }

    @media only screen and (max-width: 425px) {
      flex: 0 100%;
    }

    .playlist-card__overlay {
      align-items: center;
      display: flex;
      justify-content: center;
      position: absolute;
      height: 100%;
      transition: opacity 0.2s ease-in;
      width: 100%;
      opacity: 0;

      &--paint {
        background: ${GRAY};
        border-radius: 5px;
        height: 100%;
        opacity: 30%;
        position: absolute;
        width: 100%;
      }

      img {
        width: 50%;
        z-index: 10;
      }

      &:hover {
        opacity: 1;
      }
    }
  }
`;
