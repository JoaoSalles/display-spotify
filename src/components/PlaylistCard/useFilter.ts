import { useState, useEffect } from 'react';
import { filterFormatField } from 'types/filterFormatTypes';

const useFilter = () => {
    const [filterField, setFilterField] = useState<{[key: string]: filterFormatField}>({});

    function normilizeFilterFormat(fields: Array<any>) {
        const normilizedFormat: { [key: string]: filterFormatField} = {}
        fields.forEach(field => {
            normilizedFormat[field.id] = field;
        })
        return normilizedFormat;
    }

    async function getFilterFormat() {
        const response = await fetch('http://www.mocky.io/v2/5a25fade2e0000213aa90776')
        if (response.status === 200) {
            const reponseJson = await response.json();
            setFilterField(normilizeFilterFormat(reponseJson.filters));
        }
    }

    useEffect(() => {
        getFilterFormat();
    // eslint-disable-next-line
    }, []);

    return {
        filterField
    }
}

export default useFilter;