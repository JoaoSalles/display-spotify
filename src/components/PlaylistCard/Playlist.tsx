import React from "react";
import { PlaylistCardElement } from './styles';

interface Props {
}

function Playlist(props: Props) {
    return (
        <PlaylistCardElement className="container">
            <span>PlayList Results:</span>
        </PlaylistCardElement>
    );
}

export default Playlist;