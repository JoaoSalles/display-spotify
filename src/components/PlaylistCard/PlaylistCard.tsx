import React from "react";
import { PlaylistCardElement } from "./styles";
import defaultImage from "assets/default_playlist_image.jpg";
import playIcon from "assets/play-button.svg";

interface Props {
  name: string;
  description: string;
  external_urls: string;
  images: Array<{ url: string }>;
  tracks: number;
  id?: string;
}

function PlaylistCard(props: Props) {
  const { name, images, external_urls, description, tracks } = props;
  return (
    <PlaylistCardElement className="playlist-card" href={external_urls}>
      <div className="playlist-card__content">
        <img
          src={images.length > 0 ? images[0].url : defaultImage}
          alt={description}
        />
        <div className="playlist-card__content__description">
          <span className="playlist-card__content__description__title">
            {name}
          </span>
          <span className="playlist-card__content__description__tracks">
            {tracks} musicas
          </span>
        </div>
      </div>
      <div className="playlist-card__overlay">
        <div className="playlist-card__overlay--paint"></div>
        <img src={playIcon} alt="Play Icon" />
      </div>
    </PlaylistCardElement>
  );
}

export default React.memo(PlaylistCard);
