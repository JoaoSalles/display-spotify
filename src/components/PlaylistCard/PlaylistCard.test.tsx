import React from "react";
import { render, fireEvent, waitFor, RenderResult } from "@testing-library/react";
import PlaylistCard from "./PlaylistCard";

describe("Tests for Playlist card", () => {
    const defaultProps = {
        name: 'test',
        description: 'test description',
        external_urls: 'url_test',
        images: [],
        tracks: 1,
        id: '12312szdas'
    };

    const renderFilter = (
        props: { [key: string]: any } = {}
    ): RenderResult => {
        return render(
            <PlaylistCard {...defaultProps} {...props} />
        );
    }

    it("Should match snapshot", async () => {
        const { container } = renderFilter();

        expect(container).toMatchSnapshot();
    });
});
