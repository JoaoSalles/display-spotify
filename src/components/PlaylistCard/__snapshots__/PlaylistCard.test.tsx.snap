// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Tests for Playlist card Should match snapshot 1`] = `
<div>
  <a
    class="sc-bdnylx kTAkTV playlist-card"
    href="url_test"
  >
    <div
      class="playlist-card__content"
    >
      <img
        alt="test description"
        src="default_playlist_image.jpg"
      />
      <div
        class="playlist-card__content__description"
      >
        <span
          class="playlist-card__content__description__title"
        >
          test
        </span>
        <span
          class="playlist-card__content__description__tracks"
        >
          1
           musicas
        </span>
      </div>
    </div>
    <div
      class="playlist-card__overlay"
    >
      <div
        class="playlist-card__overlay--paint"
      />
      <img
        alt="Play Icon"
        src="play-button.svg"
      />
    </div>
  </a>
</div>
`;
